webapp/src/components/react_select_setting.jsx

```diff
@@ -95,12 +100,14 @@ export default class ReactSelectSetting extends React.PureComponent {
             );
         }
 
+        const Select = this.props.options.length > MAX_NUM_OPTIONS ? AsyncSelect : ReactSelect;
```

Comment:
Huh, interesting!

Reply:
Yes, this is not very elegant at the moment. The reason for this change is once the `AsyncSelect` component is mounted, it has chosen its "default" options as the initial options returned from the call to `loadOptions`. So if we update the `props.options`, the new options will only show up if the user types something that matches the option, because this will call `loadOptions` to match the search. When the user clicks the dropdown without any search term, only the original options passed in will show, because the `AsyncSelect` component caches those initial values.

In order to have the component be more of a "controlled" component, we need to use the non-async version of the component. Since the async version is only needed for server-side querying and large datasets (> 100 items), we need to use the default version otherwise. Do you have thoughts on how to accomplish this in a cleaner, more obvious way? @cpoile

Reply:
I think adding a comment explaining this will help us in the future, could you do that?
The other option might be to find a way to make `AsyncSelect` force reload when props.options is changed. Not sure if that would work or not, but it might make it simpler. Your choice.

Reply:
Comment sounds good. I looked into having `AsyncSelect` force reload, and couldn't find a way to do it. This is because it is meant to be used in an async context, where the only thing that matters is what the `loadOptions()` prop returns, which is not called when the user input is empty :frowning_face: